src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs

```diff
@@ -332,20 +337,22 @@ internal Location CreateLocation(LocationVersionOne v1Location)
                 {
                     FullyQualifiedLogicalName = v1Location.LogicalLocationKey ?? v1Location.FullyQualifiedLogicalName,
                     PhysicalLocation = CreatePhysicalLocation(v1Location.ResultFile),
-                    Properties = v1Location.Properties
+                    Properties = v1Location.Properties,
                 };
 
                 if (!string.IsNullOrWhiteSpace(location.FullyQualifiedLogicalName))
                 {
-                    if (_currentRun.LogicalLocations?.ContainsKey(location.FullyQualifiedLogicalName) == true)
+                    if (_v1KeyToV2LogicalLocationMap.TryGetValue(location.FullyQualifiedLogicalName, out LogicalLocation logicalLocation))
```

Comment:
>FullyQualifiedLogicalName [](start = 74, length = 25)

This is not the right lookup key. In the event of a collision, you'd have to do the lookup by means of the v1 key. So you need to do:
```
string v1Key = v1Location.LogicalLocationKey ?? v1Location.FullyQualifiedLogicalName;
```
... and pass `v1Key` to `TryGetValue`.

The value of `v1Key` is the string that was incorrectly used as the value of v2 FQLN on Line 338. Turns out we need that string after all, but for something else :-)

Reply:
[https://github.com/Microsoft/sarif-sdk/issues/1182](https://github.com/Microsoft/sarif-sdk/issues/1182)

---
In reply to: [243868340](https://github.com/Microsoft/sarif-sdk/pull/1170#discussion_r243868340) [](ancestors = 243868340)